hw/ip/otbn/dv/uvm/gen-binaries.py

```diff
@@ -0,0 +1,216 @@
+                take_env_line(env_dict, env_path, idx + 1, line)
+
+            return Toolchain(env_dict)
+    except OSError as ose:
+        raise RuntimeError('Failed to read .env file at {!r} '
+                           '(at a path {}): {}.'
+                           .format(env_path, source, ose)) from None
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--count', type=read_positive, default=10,
+                        help='Number of binaries to generate')
+    parser.add_argument('--obj-dir',
+                        help=('Object directory configured with Meson (used '
+                              'to find tool configuration). If not supplied, '
+                              'defaults to the OBJ_DIR environment variable '
+                              'if set, or build-out at the top of the '
+                              'repository if not.'))
+    parser.add_argument('--seed', type=read_positive, default=0)
+    parser.add_argument('--verbose', '-v', action='store_true')
+    parser.add_argument('--jobs', '-j', type=read_jobs, nargs='?',
+                        const='unlimited', help='Number of parallel jobs.')
+    parser.add_argument('destdir',
+                        help='Destination directory')
+
+    args = parser.parse_args()
+
+    script_dir = os.path.dirname(__file__)
+    otbn_dir = os.path.normpath(os.path.join(script_dir, '../' * 2))
+
+    try:
+        toolchain = read_toolchain(args.obj_dir, otbn_dir)
+    except RuntimeError as err:
+        print(err, file=sys.stderr)
+        return 1
+
+    rig_count = args.count - 1
+
+    os.makedirs(args.destdir, exist_ok=True)
+
+    with open(os.path.join(args.destdir, 'build.ninja'), 'w') as ninja_handle:
+        write_ninja(ninja_handle, rig_count, args.seed, toolchain, otbn_dir)
+
+    # Handle the -j argument like Make does, defaulting to 1 thread. This
+    # behaves a bit more reasonably than ninja's default (# cores) if we're
+    # running underneath something else.
+    if args.jobs is None:
+        j_arg = 1
+    elif args.jobs == 'unlimited':
+        j_arg = 0
+    else:
+        j_arg = args.jobs
+
+    cmd = ['ninja', '-j', str(j_arg)]
+    if args.verbose:
+        cmd.append('-v')
+
+    return subprocess.run(cmd, cwd=args.destdir, check=False).returncode
+
+
+def write_ninja(handle: TextIO, rig_count: int, start_seed: int,
+                toolchain: Toolchain, otbn_dir: str) -> None:
+    '''Write a build.ninja to build rig_count random binaries and a smoke test
+
+    The rules build everything in the same directory as the build.ninja file.
+    OTBN tooling is found in util_dir.
+
+    '''
+    otbn_rig = os.path.join(otbn_dir, 'util/otbn-rig')
+    smoke_src_dir = os.path.join(otbn_dir, 'dv/smoke')
+
+    seeds = [start_seed + idx for idx in range(rig_count)]
+
+    handle.write('rule rig\n'
+                 '  command = {rig} gen --size 100 --seed $seed |'
+                 ' {rig} asm -o $seed\n'
```

Comment:
Where is `$seed` set here?

Reply:
In the build rule that calls the command for a given file.

Reply:
OIC.